diff --color -ruN a/src/Cron/CronExpression.php b/src/Cron/CronExpression.php
--- a/src/Cron/CronExpression.php
+++ b/src/Cron/CronExpression.php
@@ -59,7 +59,7 @@
      *
      * @return CronExpression
      */
-    public static function factory($expression, FieldFactory $fieldFactory = null)
+    public static function factory($expression, ?FieldFactory $fieldFactory = null)
     {
         $mappings = array(
             '@yearly' => '0 0 1 1 *',
